src/toil/job.py

```diff
@@ -3026,3 +3030,32 @@ def convertPromises(kwargs):
                 foundPromisedRequirement = True
         return foundPromisedRequirement
 
+
+class JobStoreFile(str):
+    """
+    A class to wrap the job store file id returned by writeGlobalFile and any attributes we may want
+    to add to it.
+
+    It is easier to implement a startswith method here, than to inherit string because that makes
+    pickling and unpickling really hard.
+    """
+    # str isa  new-style class and we should modify __new__ instead of __init__
+    @classmethod
+    def __new__(cls, *args, **kwargs):
```

Comment:
Checkout the Promise class. I think this can be simplified by overriding __reduce__ and __new__ instead of __new__ and __getnewargs__. I don't like that class is passed twice. Also, omit **kwargs and use super() instead of hard-coding `str`.